clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareFetch.java

```diff
@@ -67,6 +70,9 @@ public void add(TopicIdPartition partition, ShareInFlightBatch<K, V> batch) {
             // but it might conceivably happen in some rare cases (such as partition leader changes).
             currentBatch.merge(batch);
         }
+        if (batch.getAcquisitionLockTimeoutMs().isPresent()) {
```

Comment:
Why do we need `isPresent` check rather just set to the one from batch?

Reply:
It's just making sure that we take the latest value, rather than overwriting an existing value with no value.